platform-tools-sdk/cargo-build-sbf/src/main.rs

```diff
@@ -517,6 +517,23 @@ fn check_undefined_symbols(config: &Config, program: &Path) {
     }
 }
 
+// Check if we have all binaries in place to execute the build command.
+// If the download failed or the binaries were somehow deleted, inform the user how to fix it.
+fn corrupted_toolchain(config: &Config) -> bool {
+    let toolchain_path = config
+        .sbf_sdk
+        .join("dependencies")
+        .join("platform-tools")
+        .join("rust");
+
+    if !toolchain_path.exists() {
+        return true;
+    }
+
+    let binaries = toolchain_path.join("bin");
+    !binaries.exists() || !binaries.join("rustc").exists() || !binaries.join("cargo").exists()
```

Comment:
nit: it might be clearer to do all of these checks as one, and it might be safer to use `try_exists()` to cover more cases https://doc.rust-lang.org/std/path/struct.Path.html#method.try_exists
```suggestion
    let binaries = toolchain_path.join("bin");
    !toolchain_path.try_exists().unwrap_or(false)
        || !binaries.try_exists().unwrap_or(false)
        || !binaries.join("rustc").try_exists().unwrap_or(false)
        || !binaries.join("cargo").try_exists().unwrap_or(false)
```

Reply:
Fixed in https://github.com/anza-xyz/agave/pull/5415/commits/f5e2f565033c5298b814dc4d7bc2df32fe8bb76b